### PR TITLE
Implemented game manager script with combat debug

### LINKED ISSUE #18: Implement game logic script

### PR DESCRIPTION
**Pull request by BenKellySoftware**
### Pull Request Description

- Added game manager and player stats savegame section
- Debug units assignment
- Debug resources list
- Debug combat simulation
- Debug action simulation

### The PR has been...
- [x] provided a reasonable name that is not just the branch name (e.g "Implemented walls mechanic")
- [x] linked to its related issue *(if any)*
- [x] assigned a reviewer from the team

### The code has been...
- [x] made mergable and free of conflicts in relation to the base branch *(according to GitHub)*
- [x] pulled to the reviewer's machine and reasonably tested
- [x] read through and approved by a reviewer
- [x] bumped in project version over the [latest release](https://github.com/CapsCollective/rng-stronghold/releases) (i.e. vX.Y.Z for major, minor or patch)

_Note: for new features, use minor (Y), and for bug fixes or small changes, use patch (Z). Bumping the minor version should also reset the patch version to zero, so v1.2.4, would become v1.3.0. See [semantic versioning](https://semver.org) for more info._

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,5 @@ mono_crash.*.json
 
 # Project export ignores
 build/
+
+*~lock~
```
Comment (BenKellySoftware) [inline .gitignore]: This is for affinity files

File: assets/levels/level_bindings.gd [hunk 2]
```diff
@@ -5,8 +5,6 @@ extends Node3D
 @export var game_cam: Camera3D
 
 func _ready():
-	Savegame.load_file()
```
Comment (BenKellySoftware) [inline assets/levels/level_bindings.gd:8]: Moved this into the GameManager so it loads before the rest of the scene initialises

File: .gitignore (hunk 1 above)
Comment (jonjondev) [reply .gitignore]: After thinking about it, the existence of this file implies that you are currently editing a file, which means that you shouldn't be messing with Git while it's being edited anyway. I'm not sure we should keep this.

File: assets/actions/bake_bread.gd [hunk 3]
```diff
@@ -0,0 +1,13 @@
+@tool
+extends BuildingAction
+
+const FLOUR_INPUT = 1
+const FOOD_OUTPUT = 3
+
+func valid_roll(roll: int):
```
Comment (jonjondev) [inline assets/actions/bake_bread.gd]: Naming does not imply predicate

**Review by jonjondev — APPROVED** [review]
Looking good to me. I've done a pass over everything, so maybe good for you to look back over and make sure it's all ok. This big contentious change was framing the actions as nodes. I think it will give us a lot of flexibility going forward.

**Review by jonjondev — APPROVED** [review]
I've done my pass on it and I'm fairly happy with the way it's setup. Passing on to you for the go-ahead.

**Comment by jonjondev** [comment]
Last major change was logically separating the concept of buildings (selection) from their actions (game logic state).

**Pull request merged**